### ISSUE TITLE
Build errors when compiling with VS 17 2022 

### ISSUE THREAD

**Issue by NickG-NZ** [Simple-Robotics/proxsuite#57]
I'm getting a massive amount of errors (syntax errors etc. see attached files) when trying to build the tests and examples on Windows 10 with MSVC in Visual Studio 17 2022. (Microsoft (R) C/C++ Optimizing Compiler Version 19.32.31329 for x64)

I'm compiling with _BUILD_TESTING_=True, _BUILD_WITH_VECTORISATION_SUPPORT_=False and _BUILD_PYTHON_INTERFACE_=False
All other cache variables are set to their defaults (except _CMAKE_PREFIX_PATH_ which is set to point to **Eigen** and **matio**).

I'm using **Eigen-3.4.0** and **matio-1.5.23** 
(As an aside, the link on [this ](https://simple-robotics.github.io/proxsuite/md_doc_5_installation.html) page confusingly points to **matio-cpp** but the **proxsuite** library actually appears to link the [original ](https://github.com/tbeu/matio) **matio** library. )
The dependencies seem to be found and linked fine. Note, I built **matio** without the optional **HDF5** dependency.

I've attached the Visual Studio Error List and the command-line output from CMake.

An example of the kind of errors I'm getting
`Error	C2504	'proxsuite::linalg::veg::tuple::IndexedTuple<proxsuite::linalg::veg::_detail::_meta::integer_sequence<proxsuite::linalg::veg::usize> *>': base class undefined	C:\Repos\proxsuite\out\build\x64-Debug\proxsuite	C:\Repos\proxsuite\include\proxsuite\linalg\veg\tuple.hpp	507	`

[ErrorList.csv](https://github.com/Simple-Robotics/proxsuite/files/9865987/ErrorList.csv)
[clOutput.txt](https://github.com/Simple-Robotics/proxsuite/files/9866000/clOutput.txt)

**Comment by fabinsch** [comment]
Hey @NickG-NZ , we know about this problem, currently there is a PR #49 ongoing that is having the some troubles when compiling examples and unittests with MSVC and we are working on a fix.

You can see the problem in your clOutput.txt in line 47, MSVC does not define a type that we use in our unittests to generate random qp:
```
C:\Repos\proxsuite\include\proxsuite\proxqp\utils\random_qp_problems.hpp(68): error C2061: syntax error: identifier '__uint128_t'
```

If you can switch to ClangCl as a compiler as we do in our [pipeline](https://github.com/Simple-Robotics/proxsuite/blob/main/.github/workflows/ci-linux-osx-win-conda.yml#L87) it works fine, otherwise you can deactivate the unittests and it should also be fine.

**Comment by jcarpent** [comment]
Solved with #49
